alf/environments/suite_babyai.py

```diff
@@ -201,19 +220,23 @@ def reset(self, **kwargs):
         return self._transform_observation(obs)
 
     def _transform_observation(self, observation):
+        # Note: The original BabyAI environment give the same instruction at every
+        # steps of an episode.
         observation['direction'] = np.int64(observation['direction'])
         mission = observation['mission']
         if not self._one_token_per_step:
-            observation['mission'] = self._tokenize(mission)
+            observation['mission'] = self._vectorize(mission)
         else:
             if mission != self._last_mission:
                 if mission != '':
```

Comment:
This check is unnecessary if the env gives the same instruction at every step?

Reply:
I am thinking of when customizing the environment, I may need to issue several different sentences separated by '' in one episode.